compiler/passes/normalize.cpp

```diff
@@ -4200,6 +4205,39 @@ static void addToWhereClause(FnSymbol*  fn,
                              ArgSymbol* formal,
                              Expr*      test);
 
+static void fixupCastFormals(FnSymbol* fn) {
+  for_formals(formal, fn) {
+    if (BlockStmt* typeExpr = formal->typeExpr) {
+      if(typeExpr->body.length == 1) {
+        if(CallExpr* typeCast = toCallExpr(typeExpr->body.tail)) {
+          if(typeCast->isCast()) {
+            VarSymbol* tmp = newTemp("call_type_tmp");
+            tmp->addFlag(FLAG_TYPE_VARIABLE);
+            tmp->addFlag(FLAG_MAYBE_TYPE);
+            tmp->addFlag(FLAG_EXPR_TEMP);
+
+            SymExpr* tmpSe = new SymExpr(tmp);
+            typeCast->replace(tmpSe);
+
+            auto mod = fn->getModule();
+            auto def = new DefExpr(tmp);
+            auto move = new CallExpr(PRIM_MOVE, tmp, typeCast);
+            mod->block->insertAtTail(def);
+
+            // insert the move just after we def the fromType
+            if(auto fromType = toSymExpr(typeCast->castFrom())) {
+              Symbol* fromTypeSym = toSymExpr(fromType)->symbol();
```

Comment:
What happens if it's not a type, in fact? E.g. `proc f(a, b: a) { }`. Could you add a test of this (which might be a .future)?

Reply:
I was a surprised to discover that `proc f(a, b: a) {}` works on main. I will add a future, since this should not compile.

Doing `proc f(a, b: a:myBorrowedType)` also works on both main and this branch, it should also not compile.

Doing `proc f(a, b: borrowed)` has the previous failure mode on main `':' undeclared`, but now seg faults. Fixing the segfault and adding all three of these as tests